Sources/Sentry/SentryHttpTransport.m

```diff
@@ -19,6 +19,13 @@
 @property (nonatomic, strong) id<SentryRateLimits> rateLimits;
 @property (nonatomic, strong) SentryEnvelopeRateLimit *envelopeRateLimit;
 
+/**
+ * Synching with a dispatch queue to have concurrent reads and writes as barrier blocks is roughly
+ * 30% slower. As this is the only part of this class that needs synchronization it is fine to sync
+ * on self.
+ */
+@property (nonatomic) BOOL isSending;
```

Comment:
Good point. I tried with a dispatch queue first, because I was afraid of using `@synchronized` or `atomic` might slow it down too much, but I found out that is actually faster. Basically using `atomic` here is roughly the same as we have right now with all the `@synchronized` keyword. I just found out again "premature optimization is the root of all evil" 😄

Reply:
lol, yeah

Also: 1 I/O equals 100000 calls (on a single thread, take contention away) to a @synchronized method.

And all that is right, or wrong, we don't know, unless we measure.